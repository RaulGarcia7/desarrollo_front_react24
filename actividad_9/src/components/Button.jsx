import styled from "styled-components";


const Button = styled.button`

    ${(props) => {
        switch (props.type) {
            case "success":
                return `
                    background-color: #17c700;
                    color: #000000;
                    padding: 10px 20px;
                    border: none;
                    border-radius: 7px;
                    cursor: pointer;

                    &:hover {
                        background-color: #066106;
                        color: whitesmoke;
                    } 
                `;
            case "error":
                return `
                    background-color: #a80e0e;
                    color: #ffffff;
                    padding: 10px 20px;
                    border: none;
                    border-radius: 7px;
                    cursor: pointer;

                    &:hover {
                        background-color: #570202;
                        color: #ffffff;
                        align-content: centre;
                        width: 250px;
                        height: 50px;
                        font-size: 30px;
                    } 
                `;

            case "warning":
                return `
                    background-color: #e4c30a;
                    color: #000000;
                    padding: 10px 20px;
                    border: none;
                    border-radius: 7px;
                    cursor: pointer;

                    &:hover {
                        background-color: #000000;
                        color: #d8cb11;
                    } 
                `;

            case "info":
                return `
                    background-color: #0a82e4;
                    color: #ffffff;
                    padding: 10px 20px;
                    border: none;
                    border-radius: 7px;
                    cursor: help;

                    &:hover {
                        background-color: #7be7de;
                        color: #000000;
                    } 
                `;
            default:
                return `
                background-color: #6cccac;
                color: black;
                padding: 10px 20px;
                border: none;
                border-radius: 7px;
                cursor: pointer;

                &:hover {
                    background-color: #2b3ff0;
                    color: whitesmoke;
                } 
            `
                
        }
    }}
`;

export default Button;