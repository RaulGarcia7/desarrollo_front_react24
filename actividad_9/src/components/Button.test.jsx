import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Button from './Button';

describe('<Button />', () => {
    it("should match snapshot without props", () => {
        render(<Button />);
        expect(screen.getByRole("button")).toMatchSnapshot();
    });
    

    it("should match snapshots with type='success'", () => {
        render(<Button type='success'/>);
        expect(screen.getByRole("button")).toMatchSnapshot();
    });

    it("should match snapshot with type='error'", () => {
        render(<Button type="error" />);
        expect(screen.getByRole("button")).toMatchSnapshot();
    });
    
    it("should match snapshot with type='warning'", () => {
        render(<Button type="warning" />);
        expect(screen.getByRole("button")).toMatchSnapshot();
    });
    
    it("should match snapshot with type='info'", () => {
        render(<Button type="info" />);
        expect(screen.getByRole("button")).toMatchSnapshot();
    });
});