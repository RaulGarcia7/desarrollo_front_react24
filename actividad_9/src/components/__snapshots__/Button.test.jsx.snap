// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should match snapshot with type='error' 1`] = `
.c0 {
  background-color: #a80e0e;
  color: #ffffff;
  padding: 10px 20px;
  border: none;
  border-radius: 7px;
  cursor: pointer;
}

.c0:hover {
  background-color: #570202;
  color: #ffffff;
  align-content: centre;
  width: 250px;
  height: 50px;
  font-size: 30px;
}

<button
  class="c0"
  type="error"
/>
`;

exports[`<Button /> > should match snapshot with type='info' 1`] = `
.c0 {
  background-color: #0a82e4;
  color: #ffffff;
  padding: 10px 20px;
  border: none;
  border-radius: 7px;
  cursor: help;
}

.c0:hover {
  background-color: #7be7de;
  color: #000000;
}

<button
  class="c0"
  type="info"
/>
`;

exports[`<Button /> > should match snapshot with type='warning' 1`] = `
.c0 {
  background-color: #e4c30a;
  color: #000000;
  padding: 10px 20px;
  border: none;
  border-radius: 7px;
  cursor: pointer;
}

.c0:hover {
  background-color: #000000;
  color: #d8cb11;
}

<button
  class="c0"
  type="warning"
/>
`;

exports[`<Button /> > should match snapshot without props 1`] = `
.c0 {
  background-color: #6cccac;
  color: black;
  padding: 10px 20px;
  border: none;
  border-radius: 7px;
  cursor: pointer;
}

.c0:hover {
  background-color: #2b3ff0;
  color: whitesmoke;
}

<button
  class="c0"
/>
`;

exports[`<Button /> > should match snapshots with type='success' 1`] = `
.c0 {
  background-color: #17c700;
  color: #000000;
  padding: 10px 20px;
  border: none;
  border-radius: 7px;
  cursor: pointer;
}

.c0:hover {
  background-color: #066106;
  color: whitesmoke;
}

<button
  class="c0"
  type="success"
/>
`;
